### PR TITLE
Add tuple class member storing key of each tuple component

### PR DESCRIPTION
**Pull request by mrw8**
For every tuple class, a class variable was added that stores a mapping from each component to the key's component name. This is useful, as it allows for trivial checking if a json object contains the correct structure for any particular tuple type. 

Some other small changes include updating TempRef to be simply have one internal reference variable instead of two for whether it is a calendar or a uuid and test changes to match the TempRef change.

### CODE REVIEW

File: src/rt_core_v2/rttuple.py [hunk 1]
```diff
@@ -28,43 +30,65 @@ class TupleType(ValueEnum):
...
 
-
 """Enum representing portions of reality types"""
 class PorType(ValueEnum):
 	singular = '+SU'
 	non_singular = '-SU'
 
+class TupleComponents(enum.Enum):
+	ar = 'ar'
+	C = 'C'
```
Comment (hoganwr) [inline src/rt_core_v2/rttuple.py:45]: The variable name C is showing up in the same font color as 'class' and 'TupleComponents' and 'Enum' so I am worried about whether it has some special significance to Python.

File: src/rt_core_v2/rttuple.py [hunk 2]
```diff
@@ -76,6 +100,8 @@ class Atuple(RtTuple):
 	unique -- 
 	t -- The time of the creation of the Atuple
 	"""
+	params = {**RtTuple.params, **enum_to_dict({TupleComponents.ar, TupleComponents.t, TupleComponents.ruia, TupleComponents.unique, TupleComponents.ruip})}
```
Comment (hoganwr) [inline src/rt_core_v2/rttuple.py:103]: Will every instance of Atuple have its own identical copy of params?

**Review by hoganwr — APPROVED** [review]

File: src/rt_core_v2/rttuple.py (hunk 2 above)
Comment (mrw8) [reply src/rt_core_v2/rttuple.py:103]: Every tuple has a shared class variable that contains the parameters for the tuple type.

File: src/rt_core_v2/rttuple.py (hunk 1 above)
Comment (mrw8) [reply src/rt_core_v2/rttuple.py:45]: I'm not seeing this as an issue in my vscode. It could be an issue with github's highlighting.

**Pull request merged**